pkg/route/hostassignment/assignment_test.go

```diff
@@ -52,6 +53,13 @@ func TestHostWithWildcardPolicies(t *testing.T) {
 		expected          string
 		expectedSubdomain string
 
+		// fields for externalCertificate
+		validType  bool
+		secretData map[string]string
+		validNS    bool
```

Comment:
Same as in `validation` package. What do you think about providing `corev1.Secret` as a test case input?